kernel/src/scan/mod.rs

```diff
@@ -154,13 +154,13 @@ impl ScanResult {
     }
 }
 
-/// Scan uses this to set up what kinds of columns it is scanning. For `Selected` we just store the
-/// name of the column, as that's all that's needed during the actual query. For `Partition` we
-/// store an index into the logical schema for this query since later we need the data type as well
-/// to materialize the partition column.
+/// Scan uses this to set up what kinds of top-level columns it is scanning. For `Selected` we just
+/// store the name of the column, as that's all that's needed during the actual query. For
+/// `Partition` we store an index into the logical schema for this query since later we need the
+/// data type as well to materialize the partition column.
 pub enum ColumnType {
     // A column, selected from the data, as is
-    Selected(ColumnName),
```

Comment:
@nicklan -- I believe this was a mistake of mine in the prefactor PR -- `ColumnType` intentionally tracks only top-level columns. So there's never a need for nesting and a normal `String` field name suffices.